### ISSUE TITLE
crash sessions no report

### ISSUE THREAD

**Issue by NikoLXQ** [getsentry/sentry-cocoa#3181]
### Platform

iOS

### Installed

Manually

### Version

Sentry 23.7.0.dev0 f52e733

### Steps to Reproduce

1. use the demo ios-objectiveC
2. click the button crash , wait the crash, and reopen demo (retry couple times)
3. check the release or project
4. the crash free sessions always 100%

i read the source code, 
![image](https://github.com/getsentry/sentry-cocoa/assets/16128430/805da073-f1a6-466e-bd09-4aaa7beb845e)
the Integrations order will effect.

### Expected Result

have the right crash free sessions

### Actual Result

the crash free sessions always 100%

**Comment by brustolin** [comment]
Hello @NikoLXQ, thanks for pointing this out, and a big thanks for showing how to solve it. 
Although I could simply change the order of integrations to fix this, I'm trying to understand what's really going on and maybe find a solution that doesn't require a specific order of integrations. 
We should have a fix for this very soon.

**Comment by armcknight** [comment]
Hi @NikoLXQ , thanks again for reporting this. We just shipped a hotfix in 8.9.3 to put the order of integrations back to their original order. Please let us know if this unblocks you!

**Comment by NikoLXQ** [comment]
Thanks.It's work now.  Actually, I read the source code and have a suggestion. While I was integrating, I found like some features I didn't use (eg. performance analysis). For the purpose of code streamlining, since we integrate with sdk, we always want the sdk to be as small as possible. But I found that it's not easy to remove features that I don't use, and the code is not strictly feature independent. Is it possible to use the macro control, in the compilation time directly remove the useless function.


Translated with www.DeepL.com/Translator (free version)
